src/main/kotlin/calculator/domain/Number.kt

```diff
@@ -0,0 +1,45 @@
+package calculator.domain
+
+const val SIMPLE_DELIMITER = ",|:"
+val SIMPLE_DELIMITER_REGEX = Regex(SIMPLE_DELIMITER)
+val NUMERIC_REGEX = Regex("[0-9]")
+val DELIMITER_REGEX = Regex("//(.)\n(.*)")
+
+fun validate(string: String?): String {
+    if (string.isNullOrBlank()) {
+        return "0"
+    }
+    return string
+}
+
+fun matchResult(text: String): MatchResult? {
+    return DELIMITER_REGEX.find(text)
+}
+
+fun getDelimiter(matchResult: MatchResult?): String {
+    if (matchResult != null) {
+        return matchResult.groupValues[1]
+    }
+    return SIMPLE_DELIMITER
+}
+
+fun parse(text: String): List<String> {
+    val matchResult = matchResult(text)
+    val delimiter = getDelimiter(matchResult)
+
+    if (delimiter == SIMPLE_DELIMITER) {
+        return text.split(SIMPLE_DELIMITER_REGEX)
+    }
+
+    return matchResult!!.groupValues[2].split(delimiter)
+}
+
+class Number(private val number: String) {
+
+    fun isNature(): Int {
+        if (NUMERIC_REGEX.matches(number)) {
+            return number.toInt()
+        }
+        throw IllegalArgumentException("숫자가 아닙니다")
+    }
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
class Number(private val number: Int) {
    constructor(number: String) : this(number.toIntOrNull() ?: throw IllegalArgumentException("숫자가 아닙니다"))
```